let CustomErrorHandler,JwtService;_616‍.x([["default",()=>_616‍.o]]);_616‍.w("../services/CustomErrorHandler",[["default",["CustomErrorHandler"],function(v){CustomErrorHandler=v}]]);_616‍.w("../services/JwtService",[["default",["JwtService"],function(v){JwtService=v}]]);


const auth = async (req, res, next) => {
  let authHeader = req.headers.authorization
  if (!authHeader) {
    return next(CustomErrorHandler.unAuthorized('No Token'))
  }
  const token = authHeader.split(' ')[1]

  try {
    const { _id, name, email, profession } = JwtService.verify(token)
    if (!profession) {
      return next(CustomErrorHandler.unAuthorized('Only For Teachers'))
    }

    const user = {
      _id,
      name,
      email,
      profession
    }

    req.user = user
    next()
  } catch (error) {
    return next(CustomErrorHandler.unAuthorized('something Went Wrong in'))
  }
}

_616‍.d(auth);
